src/app/features/create/create.component.ts

```diff
@@ -0,0 +1,135 @@
+import { firstValueFrom } from 'rxjs';
+import { GlobalErrorService } from '@core/components/global-error/global-error.service';
+import { CustomFormControl, FormNodeTypes } from '@forms/services/dynamic-form.types';
+import { DynamicFormService } from '@forms/services/dynamic-form.service';
+import { GlobalError } from '@core/components/global-error/global-error.interface';
+
+@Component({
+  selector: 'app-create',
+  templateUrl: './create.component.html'
+})
+export class CreateComponent implements OnChanges {
+  vehicle: Partial<VehicleTechRecordModel> = {};
+
+  vehicleForm = new FormGroup({
+    vin: new CustomFormControl(
+      {
+        name: 'input-vin',
+        type: FormNodeTypes.CONTROL
+      },
+      '',
+      [Validators.minLength(3), Validators.maxLength(21), Validators.required]
+    ),
+    vrmTrm: new CustomFormControl({ name: 'input-vrm-or-trailer-id', type: FormNodeTypes.CONTROL }, '', [
+      Validators.minLength(1),
+      Validators.maxLength(9),
+      Validators.required
+    ]),
+    vehicleStatus: new CustomFormControl({ name: 'change-vehicle-status-select', type: FormNodeTypes.CONTROL }, '', [Validators.required]),
+    vehicleType: new CustomFormControl({ name: 'change-vehicle-type-select', type: FormNodeTypes.CONTROL }, '', [Validators.required])
+  });
+
+  constructor(
+    private globalErrorService: GlobalErrorService,
+    private technicalRecordService: TechnicalRecordService,
+    private route: ActivatedRoute,
+    private router: Router
+  ) {}
+
+  ngOnChanges(): void {
+    this.globalErrorService.clearErrors();
+  }
+
+  get vehicleTypeOptions(): MultiOptions {
+    return [
+      { label: 'Heavy goods vehicle (HGV)', value: VehicleTypes.HGV },
+      { label: 'Public service vehicle (PSV)', value: VehicleTypes.PSV },
+      { label: 'Trailer (TRL)', value: VehicleTypes.TRL }
+    ];
+  }
+
+  get vehicleStatusOptions(): MultiOptions {
+    return [
+      { label: 'Current', value: StatusCodes.CURRENT },
+      { label: 'Provisional', value: StatusCodes.PROVISIONAL }
+    ];
+  }
+
+  get primaryVrm(): string {
+    return this.vehicle.vrms!.find(vrm => vrm.isPrimary)!.vrm;
+  }
+
+  get isFormValid(): boolean {
+    const errors: GlobalError[] = [];
+
+    DynamicFormService.updateValidity(this.vehicleForm, errors);
+
+    this.globalErrorService.setErrors(errors);
+
+    return this.vehicleForm.valid;
+  }
+
+  navigateBack() {
+    this.globalErrorService.clearErrors();
+    this.router.navigate(['..'], { relativeTo: this.route });
+  }
+
+  async handleSubmit() {
+    if (!this.isFormValid || !(await this.isFormValueUnique())) return;
+
+    this.technicalRecordService.updateEditingTechRecord(this.vehicle as VehicleTechRecordModel);
+    this.technicalRecordService.generateEditingVehicleTechnicalRecordFromVehicleType(this.vehicle.techRecord![0].vehicleType);
+    await this.router.navigate(['../create/new-record-details'], { relativeTo: this.route });
+  }
```

Comment:
```suggestion
  handleSubmit() {
    if (!this.isFormValid || !(await this.isFormValueUnique())) return;

    this.technicalRecordService.updateEditingTechRecord(this.vehicle as VehicleTechRecordModel);
    this.technicalRecordService.generateEditingVehicleTechnicalRecordFromVehicleType(this.vehicle.techRecord![0].vehicleType);
    this.router.navigate(['../create/new-record-details'], { relativeTo: this.route });
  }
```
Not sure why you would need to await the `router.navigate`